### ISSUE TITLE
It seems unfair to rate candidate concepts

### ISSUE THREAD

**Issue by unknowed-ER** [RowitZou/CG-nAR#8]
In class ‘GraphGenerator’, you used the following statement to fuse the two scores(That is formula 8 in your paper), but this may bring some unfairness:
```
prob = ((tail_attn * head_attn.unsqueeze(-1)).sum(1)).view(bsz, step_num, -1)
```
In the process of building the concept map, it seems that you use the full connection method to build the edge relationship between multiple concepts in two adjacent sentences. This makes that for the concept prediction task, the correct concept set must be the subset of the intersection of the first-order neighborhood of the concepts in the last sentence. The above scoring code will add the scores of intersection concepts derived from multiple head concepts, making the scores of intersection concepts easier to be higher than that of complement concepts.
It seems to be a better way to introduce regularization for the score of the parent concept, that is, modify it to the following code. 'head_attn' doesn't be softmax at dim 1 in self.head_attn()
```
head_attn = head_attn.unsqueeze(-1).expand(-1, -1, tail_attn.shape[-1])
head_attn = head_attn.masked_fill(~tail_attn_mask.reshape(-1, head_num, tail_num), -1e9)
head_attn = F.softmax(head_attn, dim=1)
prob = ((tail_attn * head_attn).sum(1)).view(bsz, step_num, -1)
```
Is my understanding correct? Look forward to your reply.
